### ISSUE TITLE
Provide API to allow the user to set read timeout for HTTPS send operation

### ISSUE THREAD

**Issue by anhashia** [Azure/azure-iot-sdk-java#757]
- **OS and version used:** <VERSION> 
Windows 10

- **Java runtime used:** <VERSION> 

- **SDK version used:** <VERSION> 
Latest

# Description of the issue:

Provide API to allow the user to set the read timeout of HTTPS send in case user doesn't like the default value of 4 minutes.  Right now the value is hardcoded to **4 minutes**. 

# Code sample exhibiting the issue:

Right now the value is hardcoded to **4 minutes**.
 
azure-iot-sdk-java\device\iot-device-client\src\main\java\com\microsoft\azure\sdk\iot\device\transport\https\HttpsRequest.java

```
 if (this.connectionTimeout != 0)
        {
            connection.setReadTimeoutMillis(this.connectionTimeout);
        }

```

https://github.com/Azure/azure-iot-sdk-java/blob/master/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClientConfig.java#L30

# Console log of the issue:

# Need Support?
* Have a feature request for SDKs? Please post it on [User Voice](https://feedback.azure.com/forums/321918-azure-iot) to help us prioritize.
* Have a technical question? Ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/azure-iot-hub) with tag “azure-iot-hub”
* Need Support? Every customer with an active Azure subscription has access to support with guaranteed response time.  Consider submitting a ticket and get assistance from Microsoft support team
* Found a bug? Please help us fix it by thoroughly documenting it and filing an issue on GitHub (C, Java, .NET, Node.js, Python).

**Comment by timtay-microsoft** [comment]
The feature for this issue has been checked in, and will be released soon. Thanks for filing the issue, @anhashia !

**Comment by timtay-microsoft** [comment]
The feature for this issue has been released as of device client 1.21.0 so I'm closing this issue

Users may now use the setOption API to "SetHttpsConnectTimeout" and "SetHttpsReadTimeout"

**Comment by anhashia** [comment]
Not able to use it

Device client version used : **1.21.0**

**Example**

``
client.open();
client.setOption("SetHttpsReadTimeout", 30000);
``
 Throws following **exception** 
```
Client connection opened successfully
Device client opened successfully
Exception in thread "main" java.lang.IllegalArgumentException: optionName is unknown = SetHttpsReadTimeout
                at com.microsoft.azure.sdk.iot.device.DeviceClient.setOption(DeviceClient.java:671)
                at samples.com.microsoft.azure.sdk.iot.SendEvent.main(SendEvent.java:184)
```

**Comment by timtay-microsoft** [comment]
I see that this doesn't work for device client due to a small bug in 1.21.0. I'll get the fix checked in today, and do a release tomorrow.

**Comment by timtay-microsoft** [comment]
Okay, the fix for this issue now has been released as of device client 1.21.1, so I'll close this issue

**Comment by az-iot-builder-01** [comment]
@timtay-microsoft, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
